app/models/emis_redis/military_information.rb

```diff
@@ -185,7 +185,7 @@ def military_service_episodes
 
     def service_episodes_by_date
       @service_episodes_by_date ||= lambda do
-        military_service_episodes.sort_by(&:end_date).reverse
+        military_service_episodes.sort_by { |ep| ep.end_date || Time.zone.today + 3650 }.reverse
```

Comment:
how about `DateTime::Infinity`?

Reply:
Playing around with that in rails console, it doesn't seem like it wants to sort `DateTime::Infinity.new` against other `Date` instances.

Reply:
how about `Date::Infinity`?

Reply:
Same - and looks like Date::Infinity is deprecated per apidock...I'm not sure what the `::Infinity` class is supposed to be used for but it doesn't seem like it's this.